src/commands/test_balances.rs

```diff
@@ -1,58 +1,114 @@
+use crate::components::transactions::{create_burn_transaction, create_transfer_transaction};
 use crate::components::user::User;
 use crate::prelude::info;
 use orchard::keys::Scope::External;
 use orchard::note::AssetBase;
+use zcash_primitives::transaction::Transaction;
 
-#[derive(Debug, Copy, Clone)]
-pub(crate) struct TestBalances {
-    account0: i64,
-    account1: i64,
-}
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) struct TestBalances(Vec<i64>);
 
 impl TestBalances {
-    pub(crate) fn new(account0: i64, account1: i64) -> Self {
-        TestBalances { account0, account1 }
+    pub(crate) fn add_balances(&mut self, balances: Vec<(u32, i64)>) {
```

Comment:
I'm not sure about this one. Can you explain how this works? Normally we take balances from blockchain-based db. And here you just set it manually and this may not reflect the reality. Why not just re-read balances from db?

Reply:
I am only using this function for the "expected balances" setting -- so I take the balance from the db into "expected balances", I add what is expected to happen, then mine the transaction and confirm that the expected balances match the balances I now see on the db.

Reply:
I see that it is too general a function to allow though, that's a good point. I did something different for transfer and burn (see the following):

https://github.com/QED-it/zcash_tx_tool/blob/0cbd758a01d2f483f73f2861a1eae4716609a076/src/commands/test_balances.rs#L72-L84

https://github.com/QED-it/zcash_tx_tool/blob/0cbd758a01d2f483f73f2861a1eae4716609a076/src/commands/test_balances.rs#L86-L97

I think I should do the same thing for the mining case, which is the only place I am using the `add_balances` function.

Reply:
Yeah, let's better do that

Reply:
Done.